standalone-metastore/metastore-client/src/main/java/org/apache/hadoop/hive/metastore/utils/TableFetcher.java

```diff
@@ -117,6 +118,33 @@ public List<TableName> getTables() throws Exception {
     return candidates;
   }
 
+  public List<Table> getTables() throws Exception {
+    List<Table> candidates = new ArrayList<>();
+
+    // if tableTypes is empty, then a list with single empty string has to specified to scan no tables.
+    if (tableTypes.isEmpty()) {
+      LOG.info("Table fetcher returns empty list as no table types specified");
+      return candidates;
+    }
+
+    List<String> databases = client.getDatabases(catalogName, dbPattern);
+
+    for (String db : databases) {
+      Database database = client.getDatabase(catalogName, db);
+      if (MetaStoreUtils.checkIfDbNeedsToBeSkipped(database)) {
+        LOG.debug("Skipping table under database: {}", db);
+        continue;
+      }
+      if (MetaStoreUtils.isDbBeingPlannedFailedOver(database)) {
+        LOG.info("Skipping table that belongs to database {} being failed over.", db);
+        continue;
+      }
+      List<Table> tables = client.listTablesByFilter(catalogName, db, tableFilter, -1);
```

Comment:
Can you implement Batching in it? 
client.listTablesByFilter(catalogName, db, tableFilter, -1) can potentially return a very large object if the db has large number of tables.
Please refer to TableIterable, PartitionIterable classes.